markdown/postprocessors.py

```diff
@@ -75,7 +75,8 @@ def run(self, text: str) -> str:
         """ Iterate over html stash and restore html. """
         replacements = OrderedDict()
         for i in range(self.md.htmlStash.html_counter):
-            html = self.stash_to_string(self.md.htmlStash.rawHtmlBlocks[i])
+            raw: str = self.md.htmlStash.rawHtmlBlocks[i]
+            html = self.stash_to_string(raw)
```

Comment:
Wait, so mypy doesn't allow his pattern (`funct1(funct2(value))`)? That's ridiculous.

Reply:
It's not that it doesn't allow nested calls, it's that there is another problem:

> Argument 1 to `stash_to_string` of `RawHtmlPostprocessor` has incompatible type `str | Element`; expected `str`

`self.md.htmlStash.rawHtmlBlocks[i]` is type-annotated as `str | Element`
-which is correct because `md_in_html` can indeed put an `Element` there:
https://github.com/Python-Markdown/markdown/blob/a63e6f3416f2e4fc7c2d5edb2768ceed9218df20/markdown/extensions/md_in_html.py#L166

However, `stash_to_string` is guarded by a `str` type annotation because indeed for an `Element` it will just produce garbage such as `"<Element 'a' at 0x7f55e83a8ea0>"`, and mypy rightly warns about it.

So once again there is a potential bug that mypy warns us about.
Upon looking into this though, I concluded that *maybe* there is no actual bug because `RawHtmlPostprocessor` consistently only puts strings into `htmlStash`.

For now turned it into an ignore comment.

Reply:
I think you did not read my comment very attentively because I explicitly say that there is in fact code that puts Elements into the stash.